File: ape/read_newick.R
#!/usr/bin/env Rscript 

library(ape)

start <- Sys.time() # get start time
#print(start) 

#treefile <- "../data/ben_random_tree_100k.nw"
treefile <- "../data/ben_random_tree_1mio.nw"
#print(paste("reading ",treefile))

# tree <- read.tree("../data/ben_random_tree_100k.nw")
tree <- read.tree(treefile)

end <- Sys.time()
print(end) 
duration <- difftime(end, start, units="secs")
print(paste("Internal time:", duration))

File: scikit-bio/read_newick.py
#!/usr/bin/env python3

import os
import sys
import time

from skbio import TreeNode

# Get input
treefile=sys.argv[1]
print( "reading", treefile )

# Start the clock
start = time.time()

# Run run run
tree = TreeNode.read( treefile )

# Stop the clock
end = time.time()
print( "Internal time:", end - start )

# Check result
print( "tree tips", tree.count(True) )
